tracer/tracer.go

```diff
@@ -390,12 +396,42 @@ func initializeMapsAndPrograms(kmod *kallsyms.Module, cfg *Config) (
 	}
 
 	if cfg.OffCPUThreshold > 0 {
-		if err = loadKProbeUnwinders(coll, ebpfProgs, ebpfMaps["kprobe_progs"], tailCallProgs,
+		offCPUProgs := make([]progLoaderHelper, len(tailCallProgs)+2)
+		copy(offCPUProgs, tailCallProgs)
+		offCPUProgs = append(offCPUProgs, []progLoaderHelper{
+			{
+				name:             "finish_task_switch",
+				noTailCallTarget: true,
+				enable:           true,
+			},
+			{
+				name:             "tracepoint__sched_switch",
+				noTailCallTarget: true,
+				enable:           true,
+			},
+		}...)
+		if err = loadProbeUnwinders(coll, ebpfProgs, ebpfMaps["kprobe_progs"], offCPUProgs,
 			cfg.BPFVerifierLogLevel, ebpfMaps["perf_progs"].FD()); err != nil {
 			return nil, nil, fmt.Errorf("failed to load kprobe eBPF programs: %v", err)
 		}
 	}
 
+	if len(cfg.UProbeLinks) > 0 || cfg.LoadProbe {
+		uprobeProgs := make([]progLoaderHelper, len(tailCallProgs)+1)
+		copy(uprobeProgs, tailCallProgs)
```

Comment:
Does this work if both off-cpu profiling and uprobe are enabled? This would load the kprobe flavor again...
Would it make sense to have separate `if` block to load the kprobe flavor `tailCallProgs` if either of these profiling options is enabled?